### PR TITLE
Fixes #37 - Zip distribution bundle

### LINKED ISSUE #37: TS distribution package
Issue https://github.com/jboss-eap-qe/eap-microprofile-test-suite/issues/36 discusses tagging and branching strategy, we need to discuss testsuite distribution package too.

Why:
Testsuite depends on various things like maven artifacts and docker images.
Fetching these artifacts is time-consuming and we should prepare bundle which contains all of them so there is no need to fetch multiple times (matrix jobs, various platforms).
Having a distribution package helps to mitigate the need for too complex timeouts handling discussed in issue https://github.com/jboss-eap-qe/eap-microprofile-test-suite/issues/24 

How:
 - build local repo with all maven dependencies
 - detect docker images used in TS
 - fetch docker images used in TS
 - prepare scripts to run TS with prepared local repo and docker images
 - zip it

Where:
 - in theory could be in https://github.com/jboss-eap-qe/eap-microprofile-test-suite/releases but the bundle will be too big for it
 - more realistically: internal job or artifacts repository

### PR DESCRIPTION
**Pull request by rsvoboda**
We are now able to create zip distribution bundle

Bundle creation:
```
./mp.sh dist-zip
```

Execution on target machine:
```
unzip -q eap-microprofile-test-suite-dist.zip
./mp.sh dist-run
```

FYI the indent  in mp.sh for maven commands is intentional to have better visibility for goal and the additional parameters

Please make sure your PR meets the following requirements:
- [x] Pull Request contains description for the changes
- [x] Pull Request does not include fixes for multiple issues / topics
- [x] Code compiles and tests are passing
- [x] Code is self-descriptive and/or documented

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,3 +27,22 @@ Can be used to ensure code changes are compilable, `-Djboss.home=foo` is workaro
...
+Distribution bundle contains this testsuite, pre-loaded local maven repository and dump of Docker images used in tests.
+Creation of the `eap-microprofile-test-suite-dist.zip` bundle is managed via `./mp.sh` script.
+This script assumes that all the necessary artifacts present in `${user.home}/.m2/repository/`, in other words that the TS was compiled and executed on local machine prior the bundle creation.
+
+Bundle creation:
+```
+./mp.sh dist-zip
+```
+
+Execution on target machine:
```
Comment (mnovak1) [inline README.md]: There should be added note about the need to have JDK 8+ and docker daemon/docker=>podman installed on the machine as it looks like that nothing more is needed.

File: README.md [hunk 2]
```diff
@@ -27,3 +27,22 @@ Can be used to ensure code changes are compilable, `-Djboss.home=foo` is workaro
 ```
 mvn clean verify -DskipTests -DskipITs -Djboss.home=foo
 ```
+
+## Zip distribution bundle
+Distribution bundle contains this testsuite, pre-loaded local maven repository and dump of Docker images used in tests.
+Creation of the `eap-microprofile-test-suite-dist.zip` bundle is managed via `./mp.sh` script.
```
Comment (mnovak1) [inline README.md]: `mp.sh` script could be renamed to be more self-descriptive what is its purpose/doing. Maybe something like `microprofile-testsuite-bundle.sh`.

File: README.md [hunk 3]
```diff
@@ -27,3 +27,22 @@ Can be used to ensure code changes are compilable, `-Djboss.home=foo` is workaro
...
+```
+./mp.sh dist-zip
+```
+
+Execution on target machine:
+```
+unzip -q eap-microprofile-test-suite-dist.zip
+./mp.sh dist-run
+```
+Tests output is redirected into `${TEST_FQCN}-output.txt` files in `target/surefire-reports/` directory.
```
Comment (mnovak1) [inline README.md]: Here might be explained `FQCN` is Fully Qualified Class Name as I had to think of it for a while what it is :-)

File: mp.sh [hunk 4]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+DOCKER_IMAGES=`git grep Docker.Builder 2>/dev/null | sed "s/.*Docker.Builder//g" | cut -d, -f2 | cut -d\" -f2 | sort | uniq | grep -v Invalid`
+
+case "$1" in
```
Comment (mnovak1) [inline mp.sh]: Optional: adding `help` describing what each option exactly does

File: README.md (hunk 1 above)
Comment (rsvoboda) [reply README.md]: why? 

these are TS prerequisites, not exclusive to zip distro

Created https://github.com/jboss-eap-qe/eap-microprofile-test-suite/issues/52 for it

File: README.md (hunk 3 above)
Comment (rsvoboda) [reply README.md]: FQCN is standard in Java world

File: README.md (hunk 2 above)
Comment (rsvoboda) [reply README.md]: I used short name intentionally to be efficient with typing here, many popular frameworks have 2-3 characters named tools. 

I'm open to rename it if more people want long name.

File: mp.sh (hunk 4 above)
Comment (rsvoboda) [reply mp.sh]: I'm afraid script would be unnecessarily bloated, I think usage message covers this sufficiently

```shell
*)
   echo "Usage: $0 {dist|dist-zip|dist-run|save-images|load-images|ts-maven-repo|clean}"
esac
```

Target names are hopefully clear enough, each target has up to 4 command.

File: README.md (hunk 1 above)
Comment (mnovak1) [reply README.md]: When running TS using bundle then Maven is not required to be installed on the machine, right? If yes pre-requites for running from bundle will be different and it should be pointed out.

File: mp.sh (hunk 4 above)
Comment (mnovak1) [reply mp.sh]: I had to take a look at script to see difference between dist|dist-zip and why is here save-images|load-images.

File: mp.sh [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+   MODULES_WITH_MP_PREFIX=`find . -mindepth 1 -maxdepth 1 -type d | grep 'microprofile-' | cut -d'/' -f2- | tr '\n' ','`
+   ./mvnw -s settings-dist.xml -Dmaven.repo.local=$PWD/local-repo verify       -o -pl $MODULES_WITH_MP_PREFIX -Dmaven.test.redirectTestOutputToFile=true
+   ;;
+clean)
+   rm -rf local-repo
+   rm docker-images.tar.gz
+   rm eap-microprofile-test-suite-dist.zip
+   ;;
+*)
+   echo "Usage: $0 {dist|dist-zip|dist-run|save-images|load-images|ts-maven-repo|clean}"
```
Comment (mnovak1) [inline mp.sh]: `fetch-images` is missing here

Comment (rsvoboda) [reply mp.sh]: that was intention, it's just internal detail, but wanted to have it separated

File: mp.sh (hunk 4 above)
Comment (rsvoboda) [reply mp.sh]: I think I will promote just 2 end-user facing commands

File: README.md (hunk 2 above)
Comment (rsvoboda) [reply README.md]: After f2f discussion we agreed on `mp-ts.sh` naming
Will update the PR soon

**Pull request merged**